src/ng/browser.js

```diff
@@ -131,6 +131,16 @@ function Browser(window, document, $log, $sniffer) {
     // setter
     if (url) {
       var sameState = lastHistoryState === state;
+      var trailingSlashRe = /\/$/;
+      var hasTrailingSlash = trailingSlashRe.test(url);
+
+      // Normalize the inputted URL ...
+      url = urlResolve(url).href;
+
+      // ... but keep the (lack of) trailing '/' consistent
```

Comment:
Is it possible that there was a slash initially and `urlResolve` stripped it off?

Reply:
I guess if a browser "normalizes" to have no slash at the end then yes, and I'd vote to handle that just in case. WDYT?

Reply:
It can't hurt. Let's do it :smiley:

Reply:
Although now I'm wondering if trimming this is actually a problem :thinking:  (originally I didn't trim anything, but we added this to avoid adding `/` to many tests).

The whole point of this is to make the persisted `lastBrowserUrl` normalized the same way `document.location.href` is normalized. So the comparison below and maybe also the comparison in `$location` would not fail due to the normalized value being different then the non-normalized `lastBrowserUrl`.  By removing this `/` we are making `lastBrowserUrl` different then the normalized value...

Reply:
But we are normalizing all the time, so it shouldn't matter? Or if it does, we should be able to add a test that shows this.

Reply:
I think if we strip the slash here and assign to `lastBrowserUrl`, then later we compare to `$browser.url()` (which is `location.href`, which has a slash), then maybe it will detect changes when it shouldn't?
[browser.js#L148](https://github.com/jbedard/angular.js/blob/7467049dcd09efc5397213772a2c85a488bff1db/src/ng/browser.js#L148)
[browser.js#L231](https://github.com/jbedard/angular.js/blob/7467049dcd09efc5397213772a2c85a488bff1db/src/ng/browser.js#L231)

I won't change anything unless I can prove it with a test though...

Reply:
I think we already have a test that proves it actually (but I'll try to add more).

Notice when I updated the `MockWindow` to use `aElement.href` to normalize the `window.location.href` mock... [one test](https://github.com/jbedard/angular.js/commit/21263c9c546cf628bc25be450bd18079d3a66f18#diff-c41da1c6661f19ace0ef7690f777fbcfR695) also had it's input changed otherwise it failed. I think the input shouldn't have to change and the fact that it failed shows what I'm talking about here... if we are comparing `lastBrowserURL` to `$browser.url()` then they must be normalized the same way. `$browser.url()` is `location.href`, so `lastBrowserURL` should be normalized just like `location.href`.

Reply:
Yeah. I think we must keep the browser-normalized URL, slash or not.

See the new tests in the latest commit, 4 fail with the slash-removal logic (and they are run twice, so 8 failures total). Without the `urlResolve` there are 14 failures.